### PR TITLE
Group commands in functions

### PR DESCRIPTION
**Pull request by LuisCoralesM**
Now all the commands in the bash file are grouped in individual functions

### CODE REVIEW

File: provisioner.sh [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
-microk8s kubectl apply -f ./namespaces.yml
-microk8s kubectl apply -f ./yaml
-
-# Step 5. Create a local Docker container registry 
-docker run -d -p 5000:5000  --restart=always --name registry registry:2
\ No newline at end of file
+set -e
+
+function install_microk8s () {
+  # Install microk8s
```
Comment (dbalseiro) [inline provisioner.sh]: While it's good to document the code, this comment in particular doesn't add much value. In fact is a direct transcription of the function name.

https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/

File: provisioner.sh [hunk 2]
```diff
@@ -1,26 +1,49 @@
...
+  sudo snap install microk8s --classic
+
+  # Start the service with all modules enabled
```
Comment (dbalseiro) [inline provisioner.sh]: for instances, this comment is ok

File: provisioner.sh [hunk 3]
```diff
@@ -1,26 +1,49 @@
...
+  microk8s status --wait-ready
+  microk8s enable dns dashboard storage registry
+  microk8s start
+}
+
+function microk8s_permissions () {
+  # Add all the necessary packages and permissions to use microk8s
```
Comment (dbalseiro) [inline provisioner.sh]: A better comment would have been something explaining **why** do we need those permissions or what happens if we don't have the correct permissions

File: provisioner.sh [hunk 4]
```diff
@@ -1,26 +1,49 @@
...
+  sudo usermod -a -G microk8s vagrant
+  sudo chown -f -R vagrant ~/.kube
+  newgrp microk8s
+}
+
+function install_openfaas () {
+  # Install openfaas-cli
```
Comment (dbalseiro) [inline provisioner.sh]: same as the first comment.. I would've called the function `install_openfaas_cli` if that's what it does

File: provisioner.sh [hunk 5]
```diff
@@ -1,26 +1,49 @@
...
+  curl -sSL https://cli.openfaas.com | sudo -E sh
+}
+
+function clone_faasnetes () {
+  # Clone the faasnetes git repo in order to create the functions
+  git clone https://github.com/openfaas/faas-netes
+  cd faas-netes
+}
```
Comment (dbalseiro) [inline provisioner.sh]: this one seem closely related to the next one... it's probably better to merge them together or make one call the other

This has to do with [cohesion](https://thevaluable.dev/cohesion-coupling-guide-examples/)

**Review by dbalseiro — CHANGES_REQUESTED** [review]

**Comment by LuisCoralesM** [comment]
I deleted the comments related to the functions names and improved the explanation on the requested ones

**Review by dbalseiro — APPROVED** [review]
LGTM

**Pull request merged**